pkg/gpu/consumer.go

```diff
@@ -110,7 +113,12 @@ func (c *cudaEventConsumer) Start() {
 				key := streamKey{pid: pid, stream: header.Stream_id}
 
 				if _, ok := c.streamHandlers[key]; !ok {
-					c.streamHandlers[key] = newStreamHandler(key.pid, c.sysCtx)
+					cgroup := unix.ByteSliceToString(header.Cgroup[:])
+					containerID, err := cgroups.ContainerFilter("", cgroup)
+					if err != nil {
+						log.Errorf("error getting container ID for cgroup %s: %s", cgroup, err)
+					}
+					c.streamHandlers[key] = newStreamHandler(key.pid, containerID, c.sysCtx)
```

Comment:
consider introducing a helper method that returns `(*StreamHandler, error)`

Reply:
I think I have this refactored in the multi-gpu PR, I'll improve that there as we need the GPU UUID too, that way I avoid rebase conflicts.